### PR TITLE
Feat/graphql

### PR DESCRIPTION
**Pull request by gregfrasco**
# Description

> Use Graphql to login a user and redirect them to project page. Pulls down project info and theme.
Example:
http://127.0.0.1:5173/#/?projectId=8fa6c3d8-8e2c-4709-8c5d-43cee7702523
image

After Sign in it redirects me here: https://sail.codes/?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Ijk1MjJmZWUxLTg2MGItNGZlMC04MzY4LWE3OWY2MDMwMGU5OCIsInByb2plY3RJZCI6IjhmYTZjM2Q4LThlMmMtNDcwOS04YzVkLTQzY2VlNzcwMjUyMyIsInJvbGUiOjAsImlhdCI6MTY3MTU1Mjg0NCwiZXhwIjoxNjcxNjE3NjQ0fQ.gmX1WyK535KVNaiYOwYLWqruglascc7XRoHeFOFaIjg

Example of project with custom theme
http://127.0.0.1:5173/#/?projectId=8fa6c3d8-8e2c-4709-8c5d-43cee7702524
image

## Checklist

- [x] This PR can be reviewed in under 30 minutes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have assigned reviewers to this PR.

### CODE REVIEW

File: packages/client/src/graphql/graphql.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
+/* Generated File DO NOT EDIT. */
```
Comment (cbolles) [inline packages/client/src/graphql/graphql.ts:1]: NIT: if this is generated, should this be added to .gitignore and ensure the generation takes place before running dev, etc?

File: packages/client/src/graphql/auth/auth.graphql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+mutation loginEmail($projectId: String! $email: String!, $password: String!) {
```
Comment (cbolles) [inline packages/client/src/graphql/auth/auth.graphql:1]: NIT: Is this file used? Looks like the mutation is replicated in the `auth.ts`

**Review by cbolles — COMMENTED** [review]
Just had a few questions

File: packages/client/src/graphql/graphql.ts (hunk 1 above)
Comment (gregfrasco) [reply packages/client/src/graphql/graphql.ts:1]: This file is imported. I use it in the project context cause it has the Project Model in it. You don't have to have the latest up to date scheme from the server, as long as there are no breaking changes. So it also help to find out when the last update was. This file is usually you don't look at.

File: packages/client/src/graphql/auth/auth.graphql (hunk 2 above)
Comment (gregfrasco) [reply packages/client/src/graphql/auth/auth.graphql:1]: It is the src of the graphql hooks. You write the graphql you want in any graphql file and then run`npm run introspection` to produced the corresponding ts files.

**Review by cbolles — APPROVED** [review]

**Pull request merged**